.github/workflows/chart.yaml

```diff
@@ -0,0 +1,29 @@
+name: publish_helm_chart
+
+on:
+  push:
+    branches:
+      - main
+      - master
+    paths:
+      - ".github/workflows/chart.yaml"
+      - "charts/**"    
+
+jobs:
+  deploy:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: true
+          fetch-depth: 0
+      - name: Publish Helm chart
+        # pinning to the sha f1701eb82e4d4b82016e7965501c8b6d79feaec9 from https://github.com/stefanprodan/helm-gh-pages/releases/tag/v1.4.1
+        uses: stefanprodan/helm-gh-pages@f1701eb82e4d4b82016e7965501c8b6d79feaec9
+        with:
+          # GitHub automatically creates a GITHUB_TOKEN secret to use in your workflow. You can use the GITHUB_TOKEN to authenticate in a workflow run.
+          # See https://docs.github.com/en/actions/reference/authentication-in-a-workflow#about-the-github_token-secret
```

Comment:
Ah https://github.blog/changelog/2021-04-20-github-actions-control-permissions-for-github_token/

Could we configure this so that this workflow yaml explicitly needs the write permissions, then the repo level i think we can change the default to only read permissions

Reply:
@tam7t good find!

Here is a sample run with setting `contents: write` permission: https://github.com/aramase/secrets-store-csi-driver/actions/runs/1144718022.

I've also updated the workflow permissions to `Read repository contents permission`